source/fundamentals/promises.txt

```diff
@@ -0,0 +1,194 @@
+The first parameter is the method that gets called when the Promise reaches
+the **Fulfilled** state and the optional second parameter is the method that
+gets called when it reaches the **Rejected** state. These methods are called
+**callbacks** because they execute only after the Promise completes.
+
+.. code-block:: js
+
+   ...
+   collection.updateOne({ name: "Mount McKinley" }, { meters: 6190 })
+     .then(
+       res => console.log(`Updated ${res.result.n} documents`),
+       err => console.error(`Something went wrong: ${err}`)
+     );
+
+If you only need to handle transitions to the **Rejected** state, use the
+``catch()`` method which accepts a single callback, executed when the Promise
+transitions to the **Rejected** state.
+
+.. code-block:: js
+
+     const deleteResult = await collection.deleteOne({ name: "Mount Doom" })
+       .catch(err => console.error(`Fatal error occurred: ${err}`));
+
+.. note::
+
+   Certain methods in the driver such as ``find()`` return a ``Cursor``
+   instead of a Promise. To determine what type each method returns, refer to
+   the :node-api:`NodeJS API documentation <>`.
+
+Callbacks
+---------
+
+A callback is a method that gets called after another method has
+finished executing. This allows the enclosing method to continue to execute
+other commands until the original operation completes. Callbacks are often
+used to enforce the order of processing commands.
+
+In the MongoDB NodeJS driver, you can optionally declare a callback method to
+async operations that normally return Promises such as ``findOneAndUpdate()``,
+``countDocuments()``, or ``update()``. Once the operation completes execution,
+the callback method executes as shown in the following code snippet:
+
+.. code-block:: js
+
+   collection.findOneAndUpdate({ name: "Barronette Peak" }, { name: "Baronette Peak" },
+     function(error, result) {
+       if (!error) {
+         console.log(`Operation completed successfully: ${result.ok}`);
+       } else {
+         console.log(`An error occurred: ${error}`);
+       }
+    });
+
+For more information on the callback method signature for the specific
+driver method, see the :node-api:`API documentation <>`.
+
+ .. note::
+
+    If you specify a callback, the method *does not* return a Promise.
+
+Operational Considerations
+--------------------------
+
+When using ``async`` methods, use ``await`` when you call methods
+that return Promises to ensure they complete before executing any other
+logic that depends on the result. Consider the following example in which
+we iterate over a cursor using ``hasNext()``, which returns a Promise that
+resolves to a boolean that indicates whether additional results exist, and
+``next()`` which returns a Promise that resolves to the next entry the
+cursor is pointing to.
+
+One common mistake is to forget to use ``await`` to get the value of the
+result rather than the Promise object. Consider the following code:
```

Comment:
This feels like a rehash of what is being said in the above paragraph and interrupt the flow of this section to me